eodag/plugins/search/qssearch.py

```diff
@@ -1075,8 +1085,44 @@ def do_search(
             ):
                 del prep.total_items_nb
             if items_per_page is not None and len(results) == items_per_page:
-                return results
-        return results
+
+                raw_search_results = self._build_raw_search_results(
+                    results, resp_as_json, kwargs, items_per_page
+                )
+                return raw_search_results
+
+        raw_search_results = self._build_raw_search_results(
+            results, resp_as_json, kwargs, items_per_page
+        )
+        return raw_search_results
+
+    def _build_raw_search_results(self, results, resp_as_json, kwargs, items_per_page):
+        raw_search_results = RawSearchResult(results)
+        raw_search_results.search_params = kwargs | {"items_per_page": items_per_page}
+        if self.config.pagination.get("next_page_url_tpl") is not None:
+            href = self.config.pagination["next_page_query_obj_key_path"].find(
+                resp_as_json
+            )
+            next_page_token_key = (
+                unquote(self.config.pagination["parse_url_key"])
+                if "parse_url_key" in self.config.pagination
+                else self.config.pagination.get("next_page_token_key")
+            )
+            if next_page_token_key in unquote(href[0].value):
+                from urllib.parse import parse_qs, urlparse
+
+                query = urlparse(href[0].value).query
+                page_param = parse_qs(query).get(next_page_token_key)
+                if page_param:
+                    raw_search_results.next_page_token = page_param[0]
+
+        else:
+            token = self.config.pagination["next_page_query_obj_key_path"].find(
+                resp_as_json
+            )
+            raw_search_results.next_page_token = token[0].value
```

Comment:
I get `IndexError: list index out of range`  when searching on `earth_search` when calling `next_page()` on a result that already has all available products on first page